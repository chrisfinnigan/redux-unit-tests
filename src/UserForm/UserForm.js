import React, { Component } from "react";
import { connect } from "react-redux";
import { userFormAction } from "../actions/userFormAction";

export class UserForm extends Component {
  formInputHandler = event => {
    const inputData = {
      field: event.target.name,
      value: event.target.value
    };
    this.props.userFormAction(inputData);
  };
  render() {
    return (
      <form>
        <label>
          First Name
          <input
            type="text"
            id="first_name"
            name="first_name"
            value={this.props.userConfig.first_name}
            onChange={this.formInputHandler}
          />
        </label>
        <br />
        <label>
          Last Name
          <input
            type="text"
            name="last_name"
            value={this.props.userConfig.last_name}
            onChange={this.formInputHandler}
          />
        </label>
        <br />
        <label>
          Age
          <input
            type="text"
            name="age"
            value={this.props.userConfig.age}
            onChange={this.formInputHandler}
          />
        </label>
      </form>
    );
  }
}

const mapStateToProps = state => ({
  ...state
});

const mapDispatchToProps = dispatch => ({
  userFormAction: value => dispatch(userFormAction(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(UserForm);
