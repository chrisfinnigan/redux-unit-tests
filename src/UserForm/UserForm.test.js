import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { UserForm } from "./UserForm";

describe("UserForm", () => {
  const userConfig = {
    first_name: "chris",
    last_name: "finnigan",
    age: "41"
  };

  const mockUserFormActionFn = jest.fn();

  it('should render correctly in "debug" mode', () => {
    const tree = renderer
      .create(<UserForm userConfig userFormAction debug />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call userFormAction on form input change", () => {
    const wrapper = shallow(
      <UserForm userConfig userFormAction={mockUserFormActionFn} />
    );

    wrapper.find("#first_name").simulate("change", {
      target: { name: "first_name", value: "christopher" }
    });

    expect(mockUserFormActionFn.mock.calls.length).toBe(1);
  });

  it("should pass the correct data to userFormAction on form input change", () => {
    const wrapper = shallow(
      <UserForm userConfig userFormAction={mockUserFormActionFn} />
    );

    wrapper.find("#first_name").simulate("change", {
      target: {
        name: "first_name",
        value: "christopher"
      }
    });

    expect(mockUserFormActionFn.mock.calls[1][0]).toEqual({
      field: "first_name",
      value: "christopher"
    });
  });
});
