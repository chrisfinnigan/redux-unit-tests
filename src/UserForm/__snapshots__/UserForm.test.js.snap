// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should render correctly in "debug" mode 1`] = `
<form>
  <label>
    First Name
    <input
      id="first_name"
      name="first_name"
      onChange={[Function]}
      type="text"
    />
  </label>
  <br />
  <label>
    Last Name
    <input
      name="last_name"
      onChange={[Function]}
      type="text"
    />
  </label>
  <br />
  <label>
    Age
    <input
      name="age"
      onChange={[Function]}
      type="text"
    />
  </label>
</form>
`;
